submariner/multiple-active-gateways.md

```diff
@@ -0,0 +1,100 @@
+gateway is redirected to the operational gateways. Both properties reduce the impact of a failure.
+
+## Design Details
+
+### Single Gateway
+
+In the current single gateway Submariner implementation, intra-cluster VXLAN tunnels are established by
+the non-gateway nodes to the gateway node, and each gateway establishes an inter-cluster tunnel to each
+other gateway in the clusterset (with the tunnel type determined by the cable driver in use) as shown
+below.
+
+![SingleGateway](./images/single-gateway.png)
+
+### Multiple Active Gateway Design
+
+The proposal for supporting multiple active gateway nodes is shown in the following diagram and described below.
+
+![MultiGateway](./images/multi-gateway.png)
+
+* Each non-gateway node will establish a VXLAN tunnel to each gateway node.
+* The non-gateway nodes will load-balance flows between the active gateway nodes.
+* Each gateway node will establish a tunnel to each peer gateway node.
+* The gateway nodes will load-balance flows between the active peer gateway nodes.
+* Load balancing will be performed using Linux resilient nexthop groups.
+
+### Multiple Active Gateway Packet Flow
+
+An example packet flow is shown in the following diagram and described below.
+
+![MultiGateway](./images/multi-gateway-flow.png)
+
+1. Node 1-1
+    1. Intercept packet destined for global service
+    2. Load balances flow to gateway 1-1
+    3. Hash ensures remaining packets in the connection also go to gateway 1-1
+2. Gateway 1-1
+    1. Determines packet is destined for Cluster 2 based on dest Globalnet IP
+    2. Load balances flow to Gateway 2-2
+    3. Perform SNAT from local source/port to one of the Globalnet address/ports for gateway 1-1
+    4. Hash ensures that remaining packets in the flow also go to Gateway 2-2
+3. Gateway 2-2
+    1. The packet is handed off to the local service load balancer, which performs necessary NAT
+operations and forwards the packet to chosen endpoint on Node 2-1.
+4. Node 2-1
+    1. Send response back to Gateway 2-2 (through service load balancer)
+    2. Note: it should be okay if the return path uses a different gateway, but I think the cluster
+load balancer will return it to the same node in order to maintain the NAT state.
+5. Gateway 2-2
+    1. Forward back to Gateway 1-1 over the appropriate tunnel
+    2. Note: It is important that Gateway 2-2 forwards the packet directly to Gateway 1-1 and not
+via the nexthop load balancer which may send it to Gateway 1-2. This can be accomplished by using host
+routes for the gateway IP addresses that are higher priority than the routes for the cluster Globalnet
+CIDRs which would use the nexthop load balancing.
+6. Gateway 1-1
+    1. Perform DNAT from the global address back to the local address
+    2. Send to the pod on Node 1-1 via local CNI.
+
+## Notes and Assumptions
+
+* This design assumes that Globalnet is being used. It may be possible to make it work without Globalnet if
```

Comment:
It's essential this works without globalnet.

Moreover, globalnet is an optional layer on top of core Submariner so I'd say to get it working without globalnet is higher priority.

Reply:
The design and POC now support non-Globalnet.